### PR TITLE
PHP-2279: Release automation

### PR DESCRIPTION
**Pull request by murphpdx**
This PR is based on the automation test branch, so it contains those changes as well. 
This contains two scripts.  
 - update-docker-image: Once a new agent has been released, this script will create a new docker image and push it to a new branch.
 - push-docker-image: This script will build, add tags, and push a new image to docker hub.

I've also added a template docker file and a template docker-entrypoint file. These will be used by the update-docker-image script.

### CODE REVIEW

File: Dockerfile-template [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# Copyright [2019] New Relic Corporation. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#------------------------------------------------------------------------------
+
+
+FROM alpine:3.10 AS build
+
+ENV NEWRELIC_VERSION
+ENV NEWRELIC_NAME newrelic-php5-${NEWRELIC_VERSION}-linux-musl
+ENV NEWRELIC_SHA d0b3cccf3a26ba0c266ae95aaea48a98283b1500b183cecd9218315f98e59c7d
```
Comment (pyohannes) [inline Dockerfile-template]: I guess the SHA needs to be a variable too?

File: tools/update-docker-image.bash [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+mkdir $version_short
+
+cp docker-entrypoint-template $version_short/docker-entrypoint.sh
+cd $version_short
+
+sed s/"ENV[[:space:]]NEWRELIC_VERSION/ENV NEWRELIC_VERSION v${version}"/ ../Dockerfile-template > Dockerfile
+
+git checkout -b $version_short
+git add .
+git commit -m "version bump"
```
Comment (pyohannes) [inline tools/update-docker-image.bash]: Let's include the version number in the commit message.

File: tools/update-docker-image.bash [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  echo "ERROR: A full version string must be passed in"
+  exit 1
+fi
+
+version=$1
+
+IFS='.'
+read -ra ADDR <<< "$1"
+IFS=' '
+version_short="${ADDR[0]}.${ADDR[1]}.${ADDR[2]}"
```
Comment (pyohannes) [inline tools/update-docker-image.bash]: Perfectly fine, but just throwing out another one-line solution:
```sh
version_short=$(echo $1 | awk -F "." '{print $1 "." $2 "." $3}')
```

File: tools/push-docker-image.bash [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+version=$1
+
+cd ${version} || exit 1
+
+docker build -t newrelic/php-daemon:latest -t newrelic/php-daemon:${version} .
+
+docker login --username=newrelicphp --password ${DOCKER_HUB_PASSWORD}
+
+docker push newrelic/php-daemon
```
Comment (pyohannes) [inline tools/push-docker-image.bash]: I also mentioned this in the C daemon PR: I wonder whether it's possible to just push dedicated tags here?

File: tools/update-docker-image.bash [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  echo "ERROR: A full version string must be passed in"
+  exit 1
+fi
+
+version=$1
+version_short=$(echo $1 | awk -F "." '{print $1 "." $2 "." $3}')
+
+wget -O /tmp/newrelic-php5-${version}-linux-musl.tar.gz https://download.newrelic.com/php_agent/archive/${version}/newrelic-php5-${version}-linux-musl.tar.gz;
+
+sha=$(sha256sum /tmp/newrelic-php5-${version}-linux-musl.tar.gz | awk '{print $1}')
```
Comment (pyohannes) [inline tools/update-docker-image.bash]: Ideally we'd create the SHA sum on package creation, store it somewhere and use it here. If packages on the download site get corrupted, this cycle of SHA creation and check might still file (calculating a SHA sum of an already corrupted package).

But that's definitely out-of-scope for this work, I just wanted to note it.

File: tools/update-docker-image.bash [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+mkdir $version_short
+
+cp docker-entrypoint-template $version_short/docker-entrypoint.sh
+cd $version_short
+
+sed s/"ENV[[:space:]]NEWRELIC_VERSION/ENV NEWRELIC_VERSION ${version}"/ ../Dockerfile-template > Dockerfile
+sed -i s/"ENV[[:space:]]NEWRELIC_SHA/ENV NEWRELIC_SHA ${sha}"/ Dockerfile
```
Comment (pyohannes) [inline tools/update-docker-image.bash]: Also a very minor nit: you can put both sed expressions in one invocation.
```
sed \
   -e s/"ENV[[:space:]]NEWRELIC_VERSION/ENV NEWRELIC_VERSION ${version}"/ \
   -e s/"ENV[[:space:]]NEWRELIC_SHA/ENV NEWRELIC_SHA ${sha}"/ \
    ../Dockerfile-template > Dockerfile
```

File: tools/push-docker-image.bash (hunk 4 above)
Comment (murphpdx) [reply tools/push-docker-image.bash]: I don't know that we can do that. I can clear all the images and tags at the beginning of the script though.

Comment (pyohannes) [reply tools/push-docker-image.bash]: 👍

**Pull request merged**